File: C Sharp/symmetricorder.cs
using System;

public class symmetricorder {
    public static void Main (string[] args) {
        for (int set = 1;; set++) {
            int cases = int.Parse (Console.ReadLine ());
            if (cases == 0)
                break;
            Console.WriteLine ("SET " + set);
            string[] names = new string[cases];
            int x = 0;
            for (int y = names.Length - 1; x < cases / 2; x++, y--) {
                names[x] = Console.ReadLine ();
                names[y] = Console.ReadLine ();
            }
            if ((cases & 1) != 0)
                names[x] = Console.ReadLine ();
            foreach (string s in names)
                Console.WriteLine (s);
        }
    }
}

File: Java/symmetricorder.java

// https://open.kattis.com/problems/symmetricorder
import java.util.Scanner;

public class symmetricorder {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        int set = 1;
        while (true) {
            int cases = sc.nextInt();
            if (cases == 0)
                break;
            sc.nextLine();
            String[] names = new String[cases / 2];
            System.out.println("SET " + set);

            for (int x = names.length - 1; x >= 0; x--) {
                System.out.println(sc.nextLine());
                names[x] = sc.nextLine();
            }

            if (cases % 2 == 1)
                System.out.println(sc.nextLine());

            for (String s : names)
                System.out.println(s);

            set++;
        }
        sc.close();
    }
}